File: video-club-rental/python/tests/title_builder.py
from __future__ import annotations

from video_club_rental import Title


class TitleBuilder:
    def __init__(self) -> None:
        self._name = "The Godfather"
        self._copies = 3

    def named(self, name: str) -> "TitleBuilder":
        self._name = name
        return self

    def with_copies(self, copies: int) -> "TitleBuilder":
        self._copies = copies
        return self

    def build(self) -> Title:
        return Title(self._name, self._copies)

File: video-club-rental/typescript/tests/TitleBuilder.ts
import { Title } from '../src/Title.js';

export class TitleBuilder {
  private _name = 'The Godfather';
  private _copies = 3;

  named(name: string): this { this._name = name; return this; }
  withCopies(copies: number): this { this._copies = copies; return this; }

  build(): Title { return new Title(this._name, this._copies); }
}
